src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -3020,6 +3020,7 @@ public void visitLambda(final JCLambda that) {
             if (needsRecovery) {
                 attribTree(that, env, recoveryInfo);
             }
+            return ; //wrong
```

Comment:
Jackpot:
warning: The 'return' statement in the 'finally' block discards unhandled exceptions